.devcontainer/devcontainer.json

```diff
@@ -5,33 +5,34 @@
     "context": ".."
   },
   "appPort": "8124:8123",
-  "context": "..",
   "postCreateCommand": "script/bootstrap",
-  "extensions": [
-    "github.vscode-pull-request-github",
-    "dbaeumer.vscode-eslint",
-    "ms-vscode.vscode-typescript-tslint-plugin",
-    "esbenp.prettier-vscode",
-    "bierner.lit-html",
-    "runem.lit-plugin",
-    "ms-python.vscode-pylance"
-  ],
   "containerEnv": {
     "WORKSPACE_DIRECTORY": "${containerWorkspaceFolder}"
   },
-  "settings": {
-    "terminal.integrated.shell.linux": "/bin/bash",
-    "files.eol": "\n",
-    "editor.tabSize": 2,
-    "editor.formatOnPaste": false,
-    "editor.formatOnSave": true,
-    "editor.formatOnType": true,
-    "[typescript]": {
-      "editor.defaultFormatter": "esbenp.prettier-vscode"
-    },
-    "[javascript]": {
-      "editor.defaultFormatter": "esbenp.prettier-vscode"
-    },
-    "files.trimTrailingWhitespace": true
+  "customizations": {
+    "vscode": {
+      "extensions": [
+        "dbaeumer.vscode-eslint",
+        "esbenp.prettier-vscode",
+        "runem.lit-plugin",
+        "github.vscode-pull-request-github",
+        "eamodio.gitlens"
```

Comment:
GitLens is a commercial extension, with freemium features. They have been moving more stuff into the paywall lately.

Additionally, I highly recommend on disabling welcome screens and such for this extension by default in the main configuration. In the addition, in its current form, will cause large introductions and welcome screens from GitLens when setting up your dev environment the first time (or when loading up a dev container in code spaces).

IMHO the latter is really not wished for, as the user gets services irrelevant information on startup, instead of the VSCode welcome; which above all is aimed at selling a commercial product.

../Frenck

Reply:
https://github.com/home-assistant/frontend/pull/14709